eo-maven-plugin/src/main/java/org/eolang/maven/Transpiler.java

```diff
@@ -151,34 +178,131 @@ private static void saveJava(final XML java, final Path generated) throws IOExce
     }
 
     /**
-     * Check for errors.
-     *
-     * @param xml The XML output
-     * @param name Name of the program
-     * @return The same XML if no errors
+     * XML validated for errors.
+     * @since 1.0
      */
-    private XML noErrors(final XML xml, final String name) {
-        final List<XML> errors = xml.nodes("/program/errors/error");
-        for (final XML error : errors) {
-            Logger.error(
+    private static class EnsureNoErrors {
+
+        /**
+         * XML to validate.
+         */
+        private final XML xml;
+
+        /**
+         * Program name.
+         */
+        private final String program;
+
+        /**
+         * Severities to fail on.
+         */
+        private final Set<ErrorSeverity> failsvts;
+
+        /**
+         * Ctor.
+         * @param xml XML to validate
+         * @param program Program name to assign with errors
+         * @param svts Set of severities to fail on
+         */
+        EnsureNoErrors(final XML xml, final String program, final Set<ErrorSeverity> svts) {
+            this.xml = xml;
+            this.program = program;
+            this.failsvts = EnumSet.copyOf(svts);
+        }
+
+        /**
+         * Validate XML. Raise exceptions if needed.
+         * @return Original XML
+         */
+        public XML validate() {
+            final List<XML> errors = this.xml.nodes("/program/errors/error");
+            for (final XML error : errors) {
+                final ErrorSeverity errsvty = severity(error);
+                if (errsvty == ErrorSeverity.ERROR) {
+                    this.log(error, Logger::error);
+                } else if (
+                    EnumSet.of(
+                        ErrorSeverity.WARNING,
+                        ErrorSeverity.ADVICE
+                    ).contains(errsvty)) {
+                    this.log(error, Logger::warn);
+                }
+            }
+            this.failOnErrors(
+                errors.stream()
+                    .collect(Collectors.groupingBy(EnsureNoErrors::severity))
+            );
+            return this.xml;
+        }
+
+        /**
+         * Extract severity from error node.
+         * @param error Error node XML
+         * @return Extracted severity
+         */
+        private static ErrorSeverity severity(final XML error) {
+            String severity;
+            try {
+                severity = error.xpath("@severity").get(0);
+            } catch (final IndexOutOfBoundsException ex) {
```

Comment:
@mximp bad idea, it's called using exceptions for flow control: https://softwareengineering.stackexchange.com/questions/189222/are-exceptions-as-control-flow-considered-a-serious-antipattern-if-so-why